src/mahoji/commands/tools.ts

```diff
@@ -312,55 +312,56 @@ async function xpGains(interval: string, skill?: string, ironmanOnly?: boolean)
 }
 
 async function kcGains(interval: string, monsterName: string, ironmanOnly?: boolean): CommandResponse {
-	let intervalValue = '';
-
-	switch (interval.toLowerCase()) {
-		case INTERVAL_DAY:
-			intervalValue = 'day';
-			break;
-		case INTERVAL_WEEK:
-			intervalValue = 'week';
-			break;
-		case INTERVAL_MONTH:
-			intervalValue = 'month';
-			break;
-		default:
-			return 'Invalid time interval.';
+	const intervalValue = {
+		[INTERVAL_DAY]: 'day',
+		[INTERVAL_WEEK]: 'week',
+		[INTERVAL_MONTH]: 'month'
+	}[interval.toLowerCase()];
+
+	if (!intervalValue) {
+		return 'Invalid time interval.';
 	}
-	const monster = killableMonsters.find(
+
+	const monster = effectiveMonsters.find(
 		k => stringMatches(k.name, monsterName) || k.aliases.some(a => stringMatches(a, monsterName))
 	);
 
 	if (!monster) {
 		return 'Invalid monster.';
 	}
 
+	const queryActivityType = killableMonsters.some(k => k.name === monster.name)
+		? `'MonsterKilling' AND (a."data"->>'mi')::int = ${monster.id}`
+		: `'${monster.name.replace(/\s+/g, '')}'`;
+
 	const query = `
-    SELECT a.user_id::text, SUM((a."data"->>'q')::int) AS qty, MAX(a.finish_date) AS lastDate 
-    FROM activity a
-    JOIN users u ON a.user_id::text = u.id
-    WHERE a.type = 'MonsterKilling' AND (a."data"->>'mi')::int = ${monster.id}
-    AND a.finish_date >= now() - interval '1 ${intervalValue}'  -- Corrected interval usage
-    AND a.completed = true
-    ${ironmanOnly ? ' AND u."minion.ironman" = true' : ''}
-    GROUP BY a.user_id
-    ORDER BY qty DESC, lastDate ASC
-    LIMIT 10`;
+	SELECT a.user_id::text, 
+		   SUM(COALESCE((a."data"->>'q')::int, (a."data"->>'quantity')::int, (a."data"->>'qty')::int)) AS qty, 
+		   MAX(a.finish_date) AS lastDate 
+	FROM activity a
+	JOIN users u ON a.user_id::text = u.id
+	WHERE a.type = ${queryActivityType}
+	AND a.finish_date >= now() - interval '1 ${intervalValue}'
+	AND a.completed = true
+	${ironmanOnly ? ' AND u."minion.ironman" = true' : ''}
+	GROUP BY a.user_id
+	ORDER BY qty DESC, lastDate ASC
+	LIMIT 10`;
+
```

Comment:
Did you purposely change all these lines?

Reply:
yes, was just trying to help readability since the Select line was very long. I also changed the spaces to tabs at the start of each line. I restored it back to how it was before though.